### PR TITLE
docs: make distinct sections more visible

### PR DESCRIPTION
**Pull request by JacobCoffee**
### Pull Request Checklist

- [ ] New code has 100% test coverage
- [ ] (If applicable) The prose documentation has been updated to reflect the changes introduced by this PR
- [ ] (If applicable) The reference documentation has been updated to reflect the changes introduced by this PR
- [ ] Pre-Commit Checks were ran and passed
- [ ] Tests were ran and passed

### Description

- Moves the help section into a menu of options
- Add in prose for topics
- Segment the TOC for the navigation so that sections are more easily visible
- Sort the TOC for usage and reference except for key sections
- Update all Discord links to use the new community link
- Lockfile to pull in latest litestar sphinx changes

### Close Issue(s)

-

### CODE REVIEW

**Review by guacs — APPROVED** [review]

File: docs/conf.py [hunk 1]
```diff
@@ -286,7 +291,23 @@
...
+                "description": "We are active on Discord with a dedicated help forum",
+                "link": "https://discord.gg/litestar-919193495116337154",
+                "icon": "coc",
+            },
+            "GitHub Discussions": {
+                "description": "You can ask questions and discuss Litestar on GitHub Discussions",
+                "link": "https://github.com/orgs/litestar-org/discussions",
+                "icon": "coc",
+            },
+            "Stack Overflow": {
```
Comment (provinzkraut) [inline docs/conf.py:305]: There’s not really anything there so why do we need a link?

File: docs/conf.py [hunk 2]
```diff
@@ -260,12 +260,17 @@
...
                 "icon": "contributing",
             },
             "Code of Conduct": {
                 "description": "Review the etiquette for interacting with the Litestar community",
-                "link": "https://github.com/litestar-org/.github/blob/main/CODE_OF_CONDUCT.md",
+                "link": "https://github.com/litestar-org/.github?tab=coc-ov-file",
+                "icon": "coc",
+            },
+            "Security": {
+                "description": "Learn how we handle security issues in Litestar",
```
Comment (provinzkraut) [inline docs/conf.py:272]: The wording doesn’t fit here. It’s not a tutorial so "learn how.." doesn’t really make sense

File: docs/conf.py [hunk 3]
```diff
@@ -276,7 +281,7 @@
                 "icon": "org",
             },
             "Releases": {
-                "description": "Details about the Litestar release process",
+                "description": "Explore the release process for Litestar",
```
Comment (provinzkraut) [inline docs/conf.py:284]: Same as above: Not a tutorial.

File: docs/topics/index.rst [hunk 4]
```diff
@@ -1,11 +1,15 @@
 Topics
 ======
 
-A collection of articles about technical concepts, background information,
+Topics are a collection of articles about technical concepts, background information,
 design decisions and similar topics on a higher level.
 
+This can include things like the difference between synchronous and asynchronous, API design,
```
Comment (provinzkraut) [inline docs/topics/index.rst:7]:
```suggestion
This includes things like the difference between synchronous and asynchronous, API design,
```
Nitpick: Not only *can* this include things like these, it actually does

**Review by provinzkraut — CHANGES_REQUESTED** [review]
Cool! A few minor change requests :)

**Pull request merged**